ambry-router/src/main/java/com/github/ambry/router/QuotaAwareOperationController.java

```diff
@@ -221,6 +221,7 @@ private void pollQuotaExceedAllowedRequestsIfAny(List<RequestInfo> requestsToSen
     Timer.Context timer = routerMetrics.pollExceedAllowedRequestTime.time();
     List<QuotaResource> quotaResources = new ArrayList<>(requestQueue.keySet());
     Collections.shuffle(quotaResources);
+    List<QuotaResource> quotaResourcesRemoved = new ArrayList<>();
     while (!requestQueue.isEmpty()) {
       for (QuotaResource quotaResource : quotaResources) {
         RequestInfo requestInfo = requestQueue.get(quotaResource).getFirst();
```

Comment:
Thanks for catching this bug. Would doing something like below here help with the performance a little?
if(!requestQueue.containsKey(quotaResource) { continue; }

Reply:
make some change, I think this probably would be the most efficient way.